src/main/java/to/us/resume_builder/spring/controller/PDFController.java

```diff
@@ -10,43 +10,82 @@
+	 * @param url Whether the caller wants the PDF uploaded to file.io
+	 * @return The response the user requested, or possibly an error.
+	 */
 	@PostMapping("/pdf")
-	public ResponseEntity<InputStreamResource> postPdf(@RequestParam(name = "latex") String latex, @RequestParam(name = "url") boolean url) {
-		LOG.info(latex);
-		Path pdf = Path.of("./", "resume.pdf");
+	public ResponseEntity<InputStreamResource> postPdf(@RequestParam(name = "latex") String latex,
+													   @RequestParam(name = "url") boolean url) {
+		Path pdf;
+		do {
+			pdf = Path.of("./", MiscUtils.randomAlphanumericString(16) + ".pdf");
+		} while (Files.exists(pdf));
+
 		InputStreamResource rsc;
 		MediaType mediaType;
+		InputStream in;
 		try {
 			if (!ResumeExporter.export(pdf, latex)) {
-				LOG.warning("Export failed");
-				return getInternalServerErrorResponse("Export failed").build();
+				LOG.warning("PDF compilation failed");
+				return badRequest("Could not convert LaTeX to PDF").build();
 			}
+
 			LOG.info("requesting url: " + url);
 			if (url) {
+				// Upload PDF to file.io, return response
 				LOG.info("Uploading pdf to file.io");
 				String response = ResumeExporter.uploadPDF(pdf);
-				rsc = new InputStreamResource(new ByteArrayInputStream(response.getBytes()));
+				in = new ByteArrayInputStream(response.getBytes());
 				mediaType = MediaType.APPLICATION_JSON;
 			} else {
-				rsc = new InputStreamResource(new BufferedInputStream(new FileInputStream(String.valueOf(pdf))));
+				// Give the PDF in the response
+				in = new ByteArrayInputStream(String.valueOf(pdf).getBytes());
 				mediaType = MediaType.APPLICATION_PDF;
 			}
 		} catch (FileNotFoundException e) {
 			LOG.warning("could not find " + pdf.toString() + " - " + e.getMessage());
-			return getInternalServerErrorResponse("internal PDF file not found").build();
-		} catch (InterruptedException | TimeoutException e) {
+			return internalServerError("internal file not found").build();
+		} catch (InterruptedException e) {
 			LOG.warning("file.io upload failed - " + e.getMessage());
-			return getInternalServerErrorResponse("file.io upload failed").build();
+			return serviceUnavailable("file.io upload failed").build();
+		} catch (TimeoutException e) {
+			LOG.warning("file.io upload timed out - " + e.getMessage());
+			return requestTimeout("file.io upload timed out").build();
 		} catch (IOException e) {
-			LOG.warning("export failure -> " + e.getMessage());
-			return getInternalServerErrorResponse("export failed").build();
+			LOG.warning("caught IOException - " + e.getMessage());
+			return internalServerError("internal file I/O error").build();
 		}
-		return getOkResponse().contentType(mediaType).body(rsc);
+
+		// Clean up PDF file
+		try {
+			Files.deleteIfExists(pdf.toAbsolutePath());
+		} catch (IOException e) {
+			LOG.warning("could not delete pdf - " + pdf.toString());
+			return internalServerError("could not clean up resource").build();
```

Comment:
Correct me if I'm wrong, but this doesn't seem like something we should trouble the user with. We got their resource, right? Shouldn't we just send it to them and pretend nothing went wrong?